packages/typespec-go/src/tcgcadapter/adapter.ts

```diff
@@ -95,6 +95,11 @@ function fixStutteringTypeNames(sdkPackage: tcgc.SdkPackage<tcgc.SdkHttpOperatio
   // ensure that enum, client, and struct type names don't stutter
 
   const recursiveWalkClients = function(client: tcgc.SdkClientType<tcgc.SdkHttpOperation>): void {
+    // NOTE: we MUST do this before calling trimPackagePrefix to properly handle
+    // the case where the client name is the same as the package name.
+    if (!client.name.match(/Client$/)) {
```

Comment:
What if the client name is already `FooClient`? We would match that and turn it into `FooClientClient`. We should check that the name doesn't already end in `Client`.

Reply:
That's what this check does. If the name doesn't end with `Client` then append `Client`.